views-jstachio/src/main/java/io/micronaut/views/jstachio/JstachioViewResolver.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.views.jstachio;
+
+import io.jstach.jstache.JStache;
+import io.jstach.jstache.JStacheConfig;
+import io.jstach.jstache.JStachePath;
+import io.micronaut.context.BeanContext;
+import io.micronaut.context.annotation.Replaces;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.annotation.AnnotationValue;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.beans.BeanIntrospection;
+import io.micronaut.core.beans.exceptions.IntrospectionException;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.inject.BeanConfiguration;
+import io.micronaut.views.DefaultViewsResolver;
+import io.micronaut.views.ViewsResolver;
+import jakarta.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Resolves view name by checking the annotations {@link JStache} and {@link JStacheConfig} in the response body.
+ * @author Sergio del Amo
+ * @since 4.1.0
+ */
+@Requires(classes = HttpRequest.class)
+@Replaces(DefaultViewsResolver.class)
+@Singleton
+public class JstachioViewResolver implements ViewsResolver {
+    private static final Logger LOG = LoggerFactory.getLogger(JstachioViewResolver.class);
+    private static final String MEMBER_PATH = "path";
+    private static final String MEMBER_PATHING = "pathing";
+    private static final String MEMBER_PREFIX = "prefix";
+    private static final String MEMBER_SUFFIX = "suffix";
+    private static final String MEMBER_USING = "using";
+    private final BeanContext beanContext;
+    private final ConcurrentHashMap<Class<?>, String> viewForClass = new ConcurrentHashMap<>();
+
+    /**
+     *
+     * @param beanContext BeanContext
+     */
+    public JstachioViewResolver(BeanContext beanContext) {
+        this.beanContext = beanContext;
+    }
+
+    @Override
+    public Optional<String> resolveView(HttpRequest<?> request, HttpResponse<?> response) {
+        Object body = response.body();
+        return body == null ?
+            Optional.empty() :
+            Optional.ofNullable(viewForClass.computeIfAbsent(body.getClass(), c -> resolveView(c).orElse(null)));
```

Comment:
You have the body class right here! That is all JStachio needs to look up template data.

`JStachio.supportsType` and then if you need more metadata `JStachio.findTemplate`.

Reply:
Anyway if you do this approach of the resolver I think you could just do something like:

```java
// ignore null checking here
if (jstachio.supportsType(response.body().getClass()) {
  String viewName = "jstachio:" + response.body().getClass().getName();
  // update view name to class map
  return Optional.of(viewName);
}
```

Reply:
[I have deleted the view resolver](https://github.com/micronaut-projects/micronaut-views/pull/590/commits/252f79680281ae3d66f7250c86aa053526a51b5b).